price/pricing/prometheus.go

```diff
@@ -0,0 +1,46 @@
+package pricing
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"github.com/prometheus/client_golang/api"
+	promV1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+// NewPromClient creates new Prometheus API client.
+func NewPromClient(address, user, password string) promV1.API {
```

Comment:
This looks very strange, part of the metrics received through the quality oracle, but here you doing direct requests to Prometheus. I'm afraid that this discovery will be a huge monster trying to do everything internally, with its own cache, processing, aggregation, etc.